### PR TITLE
feat: added threshold to colors config

### PR DESCRIPTION
**Pull request by m0ksem**
closes https://github.com/epicmaxco/vuestic-ui/issues/2354

@aluarius, should we handle hover and focus color modificators in colors config?

### CODE REVIEW

**Comment by aluarius** [comment]
> should we handle hover and focus color modificators in colors config?

@m0ksem i'm not sure if it will work with components presets.

File: packages/ui/src/services/color-config/color-functions.ts [hunk 1]
```diff
@@ -6,14 +6,14 @@ export const colorToRgba = (color: ColorInput, opacity: number) => {
   return new ColorTranslator(color).setA(opacity).RGBA
 }
 
-export const isLightBackground = (color: ColorInput) => {
+export const isLightBackground = (color: ColorInput, threshold: number) => {
```
Comment (aluarius) [inline packages/ui/src/services/color-config/color-functions.ts:9]: Where is it used inside `isLightBackground`?

**Review by aluarius — APPROVED** [review]

File: packages/ui/src/services/color-config/color-functions.ts [hunk 2]
```diff
@@ -6,14 +6,14 @@ export const colorToRgba = (color: ColorInput, opacity: number) => {
...
   // TODO: replace with color mixin (doesn't play any role in dark theme)
   const { R, G, B } = new ColorTranslator(color)
   return Math.sqrt(R * R * 0.241 + G * G * 0.691 + B * B * 0.068) > 120
```
Comment (m0ksem) [inline packages/ui/src/services/color-config/color-functions.ts:12]:
```suggestion
  return Math.sqrt(R * R * 0.241 + G * G * 0.691 + B * B * 0.068) > threshold
```

File: packages/ui/src/services/color-config/color-functions.ts (hunk 1 above)
Comment (aluarius) [inline packages/ui/src/services/color-config/color-functions.ts:9]: Mb we want to set threshold default value to 120 here in purpose not to pass it everywhere?

**Review by aluarius — APPROVED** [review]

File: packages/ui/src/composables/useColors.ts [hunk 3]
```diff
@@ -110,6 +116,10 @@ export const useColors = () => {
       }, {})
   }
 
+  const getTextColor = (color: ColorInput, darkColor = 'textDark', lightColor = 'textLight') => {
+    return isLightBackground(color, 120) ? darkColor : lightColor
+  }
```
Comment (RVitaly1978) [inline packages/ui/src/composables/useColors.ts:121]: Can't we use `getTextColor` from `'../services/color-config/color-functions'`?

File: packages/ui/src/services/color-config/color-functions.ts [hunk 4]
```diff
@@ -6,14 +6,14 @@ export const colorToRgba = (color: ColorInput, opacity: number) => {
...
   // TODO: replace with color mixin (doesn't play any role in dark theme)
   const { R, G, B } = new ColorTranslator(color)
-  return Math.sqrt(R * R * 0.241 + G * G * 0.691 + B * B * 0.068) > 120
+  return Math.sqrt(R * R * 0.241 + G * G * 0.691 + B * B * 0.068) > threshold
 }
 
-export const getTextColor = (color: ColorInput, darkColor = 'textDark', lightColor = 'textLight') => {
-  return isLightBackground(color) ? darkColor : lightColor
+export const getTextColor = (color: ColorInput, darkColor: string, lightColor: string, threshold: number) => {
+  return isLightBackground(color, threshold) ? darkColor : lightColor
```
Comment (RVitaly1978) [inline packages/ui/src/services/color-config/color-functions.ts:16]: maybe we can set default values to `darkColor = 'textDark'`, `lightColor = 'textLight'`.
Shouldn't we use `threshold` from `globalConfig.colors.threshold` by default?

File: packages/ui/src/services/color-config/plugin/create-color-config-plugin.ts [hunk 5]
```diff
@@ -32,20 +33,17 @@ export const createColorConfigPlugin = (app: App) => {
...
+      setCSSVariable(`on-${key}`, getTextColor(newValue[key], newValue.textDark, newValue.textLight, globalConfig.value.colors!.threshold), root)
     })
   }
 
-  updateColors(globalConfig.value.colors)
+  updateColors(globalConfig.value.colors!.variables)
 
   watch(() => globalConfig.value.colors, (newValue) => {
-    updateColors(newValue)
+    updateColors(newValue?.variables)
```
Comment (RVitaly1978) [inline packages/ui/src/services/color-config/plugin/create-color-config-plugin.ts:43]: Should we be looking at changes to the entire color configuration here, or are just the `variables` enough?
`watch(() => globalConfig.value.colors.variables, (newValue) => {`

**Review by RVitaly1978 — APPROVED** [review]

File: packages/ui/src/services/color-config/color-functions.ts (hunk 4 above)
Comment (m0ksem) [reply packages/ui/src/services/color-config/color-functions.ts:16]: It is used in useColors, cuz it we can use globalConfig only in setup context.

**Pull request merged**